### PR TITLE
Convert to maven and auto-release

### PR DESCRIPTION
**Pull request by samrocketman**
- Convert from building with Gradle to Maven.
- Update plugin for continuous automated release.
- No version change to jervis library.
- https://www.jenkins.io/doc/developer/publishing/releasing-cd/

Relates to https://github.com/jenkinsci/snakeyaml-api-plugin/pull/75

I'll need to make a followup release for snakeyaml fix (after autorelease).  There will be breaking code changes for the jervis library update (jervis 1.7 -> jervis 2.0) so this should be a separate PR.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            <url>http://opensource.org/licenses/MIT</url>
+        </license>
+    </licenses>
+
+    <developers>
+        <developer>
+            <id>sag47</id>
+            <name>Sam Gleske</name>
+        </developer>
+    </developers>
```
Comment (timja) [inline pom.xml:32]: no need for this, plugin site uses repository permissions updater data now as it's more accurate

**Review by timja — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (samrocketman) [reply pom.xml:32]: Ok I will remove it.  Thanks!

Comment (samrocketman) [reply pom.xml:32]: I need to check that it compiles the groovy after sleeping on it

**Comment by samrocketman** [comment]
@timja I can't get this plugin to work in the latest version of Jenkins.

I can bootstrap the latest version of jenkins and install this plugin.  No filter traits show up in GitHub branch source plugin.

File: pom.xml (hunk 1 above)
Comment (samrocketman) [reply pom.xml]: Removed but the plugin is broken per my latest review.

**Comment by samrocketman** [comment]
I verified the jars are in the classpath via script console.

```groovy
import net.gleske.jervis.tools.AutoRelease
import net.gleske.scmfilter.credential.GraphQLTokenCredential

AutoRelease.getScriptFromTemplate('${hello} ${world}', [hello: 'goodbye', world: 'friend'])
```

I'm a bit stumped as to why the filter doesn't show up as an SCM filter trait.  I can't release it in this condition...

**Comment by samrocketman** [comment]
In the new Jar I see `META-INF/services/annotations/org.kohsuke.accmod.Restricted` which contains my messages class `net.gleske.scmfilter.impl.trait.Messages`; that seems to be missing in the old?

**Comment by samrocketman** [comment]
I guess the next thing to try is look to other plugins using maven to compile groovy like job-dsl.

**Comment by samrocketman** [comment]
Switching to GMavenPlus (which is what Job DSL plugin uses to compile groovy) fixed everything.  I also upgraded to Java bytecode to Java 1.8.  It took a bit for me to figure out how and I documented this process in a false bug report https://github.com/groovy/GMavenPlus/issues/256

**Pull request merged**